Atomic/syscontainers.py

```diff
@@ -1528,18 +1531,22 @@ def uninstall(self, name):
         if installed_files:
             RPMHostInstall.rm_add_files_to_host(installed_files, None)
 
-        if os.path.lexists("%s/%s" % (checkout, name)):
-            os.unlink("%s/%s" % (checkout, name))
-        for deploy in ["0", "1"]:
-            if os.path.exists("%s/%s.%s" % (checkout, name, deploy)):
-                shutil.rmtree("%s/%s.%s" % (checkout, name, deploy))
-
-        if os.path.exists(unitfileout):
-            os.unlink(unitfileout)
-
         if rpm_installed:
             RPMHostInstall.uninstall_rpm(rpm_installed.replace(".rpm", ""))
 
+        # Until the symlink and the deployment are in place, we can attempt the uninstall again.
+        # So treat as fatal each failure that happens before we rm the symlink.
+        symlink_path = os.path.join(checkout, name)
+        if os.path.lexists(symlink_path):
```

Comment:
Is a ``daemon-reload`` needed as well?

Reply:
that should probably happen just in case of `systemctl start container-just-uninstalled` but sure, it looks like a good idea, going to add a patch